### DISCUSSION TITLE
Does this tool support parallel computing?

### DISCUSSION THREAD

**Discussion by wjzwjz5** [campbio/decontX#13] [Q&A] [answered]
When I provide an ADT count matrix of 141*212290, it seems to take more than a week to run.
![image](https://github.com/campbio/decontX/assets/49862657/3760eb60-c632-4d9b-99dc-dd33bbdfd5c8)

Thanks for your help!

**Comment by yuan-yin-truly** [comment]
Hi @wjzwjz5 I don't think you can run it parallelly. And I agree the inference speed is the major bottleneck of the tool currently.

However, your current matrix has > 200,000 droplets, which I think might have included all droplets in the raw matrix? You may want to use just the filtered matrix which include only the cell-containing droplets.

Also, are all 200,000 droplets prepped and sequenced in one batch? If not, you can break down the matrix into batches and run on each batch. Our assumption on the contamination is based on per-batch level, so theoretically decontaminating on each batch of droplets are fine.

**Reply by wjzwjz5** [reply]
Hi, Thanks for the quick reply.  Each batch contains a varying number of cells, but these 200,000 are all filtered matrices. Actually 200,000 droplets from multiple batches and I used Seurat's merge function to merge them together. Each batch contains a varying number of cells, but these 200,000 are all filtered matrices. 
I'm trying to find out if this tool work rightly on my ADT data. There are some reason make the isotype wrongly express.

**Reply by yuan-yin-truly** [reply, marked as answer]
If each batch is library prepped separately, I think you should run each batch on DecontPro separately, then merge using Seurat. I would imagine each library prep batch to have different amount of contamination.
